main.tf

```diff
@@ -1,13 +1,254 @@
+
+  attributes = ["iam"]
+
+  context = module.this.context
+}
+
+data "aws_iam_policy_document" "iam_policy_document" {
+  statement {
+    sid       = ""
+    actions   = ["airflow:PublishMetrics"]
+    effect    = "Allow"
+    resources = ["arn:aws:airflow:${local.region}:${local.account_id}:environment/${module.this.id}"]
+  }
+
+  statement {
+    sid     = ""
+    actions = ["s3:ListAllMyBuckets"]
+    effect  = "Allow"
+    resources = [
+      var.create_s3_bucket ? module.mwaa_s3_bucket.bucket_arn : var.source_bucket_arn,
+      "${var.create_s3_bucket ? module.mwaa_s3_bucket.bucket_arn : var.source_bucket_arn}/*"
+    ]
+  }
+
+  statement {
+    sid = ""
+    actions = [
+      "s3:GetObject*",
+      "s3:GetBucket*",
+      "s3:List*"
+    ]
+    effect = "Allow"
+    resources = [
+      var.create_s3_bucket ? module.mwaa_s3_bucket.bucket_arn : var.source_bucket_arn,
+      "${var.create_s3_bucket ? module.mwaa_s3_bucket.bucket_arn : var.source_bucket_arn}/*"
+    ]
+  }
+
+  statement {
+    sid       = ""
+    actions   = ["logs:DescribeLogGroups"]
+    effect    = "Allow"
+    resources = ["*"]
+  }
+
+  statement {
+    sid = ""
+    actions = [
+      "logs:CreateLogStream",
+      "logs:CreateLogGroup",
+      "logs:PutLogEvents",
+      "logs:GetLogEvents",
+      "logs:GetLogRecord",
+      "logs:GetLogGroupFields",
+      "logs:GetQueryResults",
+      "logs:DescribeLogGroups"
+    ]
+    effect    = "Allow"
+    resources = ["arn:aws:logs:${local.region}:${local.account_id}:log-group:airflow-${module.this.id}*"]
+  }
+
+  statement {
+    sid       = ""
+    actions   = ["cloudwatch:PutMetricData"]
+    effect    = "Allow"
+    resources = ["*"]
+  }
+
+  statement {
+    sid = ""
+    actions = [
+      "sqs:ChangeMessageVisibility",
+      "sqs:DeleteMessage",
+      "sqs:GetQueueAttributes",
+      "sqs:GetQueueUrl",
+      "sqs:ReceiveMessage",
+      "sqs:SendMessage"
+    ]
+    effect    = "Allow"
+    resources = ["arn:aws:sqs:${local.region}:*:airflow-celery-*"]
+  }
+
+  statement {
+    sid = ""
+    actions = [
+      "kms:Decrypt",
+      "kms:DescribeKey",
+      "kms:GenerateDataKey*",
+      "kms:Encrypt"
+    ]
+    effect        = "Allow"
+    not_resources = ["arn:aws:kms:*:${local.account_id}:key/*"]
+    condition {
+      test     = "StringLike"
+      variable = "kms:ViaService"
+      values = [
+        "sqs.${local.region}.amazonaws.com"
+      ]
+    }
   }
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+module "mwaa_security_group" {
+  source  = "cloudposse/security-group/aws"
+  version = "0.4.2"
+
+  enabled                    = local.security_group_enabled
+  security_group_description = var.security_group_description
+  allow_all_egress           = true
+  rules                      = var.additional_security_group_rules
+  rule_matrix = [
+    {
+      source_security_group_ids = var.associated_security_group_ids
```

Comment:
will it work if `var.associated_security_group_ids` is not provided?
do we always need to provide `var.associated_security_group_ids` ?

Reply:
No default is []